### ISSUE TITLE
danger local: improve message

### ISSUE THREAD

**Issue by php-coder** [danger/danger#734]
# Report

I'm a new user and I found that the error message "No recent Pull Requests found for this repo, danger requires at least one Pull Request for the local mode." is unhelpful.

First, could you remove the stack trace? It scares me because it looks like an exceptional case and error inside of the program occurs. But it's general error actually.

Second, I didn't understand what this message mean and how to fix it. I have a repo and there are 2 PRs. After reading that messages I started to think that it won't work because these PRs were created by other users, so I created a new PR from myself. But it continues to fail.

Then I have to read the code :-( I found out that it misses PR number. Why it doesn't suggest me to specify it in the error message?

Now, I've specified the PR number and it still fails:
```console
$ danger local --use-merged-pr=533
/usr/local/lib/ruby/gems/2.3.0/gems/danger-4.2.2/lib/danger/ci_source/support/pull_request_finder.rb:31:in `check_if_any_pull_request!': Could not find the Pull Request (533) inside the git history for this repo. (RuntimeError)
	from /usr/local/lib/ruby/gems/2.3.0/gems/danger-4.2.2/lib/danger/ci_source/support/pull_request_finder.rb:15:in `call'
	from /usr/local/lib/ruby/gems/2.3.0/gems/danger-4.2.2/lib/danger/ci_source/local_git_repo.rb:100:in `found_pull_request'
	from /usr/local/lib/ruby/gems/2.3.0/gems/danger-4.2.2/lib/danger/ci_source/local_git_repo.rb:45:in `initialize'
	from /usr/local/lib/ruby/gems/2.3.0/gems/danger-4.2.2/lib/danger/danger_core/environment_manager.rb:30:in `new'
	from /usr/local/lib/ruby/gems/2.3.0/gems/danger-4.2.2/lib/danger/danger_core/environment_manager.rb:30:in `initialize'
	from /usr/local/lib/ruby/gems/2.3.0/gems/danger-4.2.2/lib/danger/commands/local.rb:46:in `new'
	from /usr/local/lib/ruby/gems/2.3.0/gems/danger-4.2.2/lib/danger/commands/local.rb:46:in `run'
	from /usr/local/lib/ruby/gems/2.3.0/gems/claide-1.0.1/lib/claide/command.rb:334:in `run'
	from /usr/local/lib/ruby/gems/2.3.0/gems/danger-4.2.2/bin/danger:5:in `<top (required)>'
	from /usr/local/bin/danger:23:in `load'
	from /usr/local/bin/danger:23:in `<main>'
```

What does it mean? "Could not find the Pull Request (533) inside the git history for this repo" What is the "git history for this repo" and why the PR isn't there? Ah, probably, I should specify PR that were merged and isn't opened?!

## What did you do?

`danger local`

## What did you expect to happen?

validates my local changes

## What happened instead?

```console
$ danger local
/usr/local/lib/ruby/gems/2.3.0/gems/danger-4.2.2/lib/danger/ci_source/support/pull_request_finder.rb:33:in `check_if_any_pull_request!': No recent Pull Requests found for this repo, danger requires at least one Pull Request for the local mode. (RuntimeError)
	from /usr/local/lib/ruby/gems/2.3.0/gems/danger-4.2.2/lib/danger/ci_source/support/pull_request_finder.rb:15:in `call'
	from /usr/local/lib/ruby/gems/2.3.0/gems/danger-4.2.2/lib/danger/ci_source/local_git_repo.rb:100:in `found_pull_request'
	from /usr/local/lib/ruby/gems/2.3.0/gems/danger-4.2.2/lib/danger/ci_source/local_git_repo.rb:45:in `initialize'
	from /usr/local/lib/ruby/gems/2.3.0/gems/danger-4.2.2/lib/danger/danger_core/environment_manager.rb:30:in `new'
	from /usr/local/lib/ruby/gems/2.3.0/gems/danger-4.2.2/lib/danger/danger_core/environment_manager.rb:30:in `initialize'
	from /usr/local/lib/ruby/gems/2.3.0/gems/danger-4.2.2/lib/danger/commands/local.rb:46:in `new'
	from /usr/local/lib/ruby/gems/2.3.0/gems/danger-4.2.2/lib/danger/commands/local.rb:46:in `run'
	from /usr/local/lib/ruby/gems/2.3.0/gems/claide-1.0.1/lib/claide/command.rb:334:in `run'
	from /usr/local/lib/ruby/gems/2.3.0/gems/danger-4.2.2/bin/danger:5:in `<top (required)>'
	from /usr/local/bin/danger:23:in `load'
	from /usr/local/bin/danger:23:in `<main>'
```

## Your Environment

```console
$ danger --version
4.2.2
```

**Comment by php-coder** [comment]
Looks like danger looking for commits with the format `Merge pull request #xxx`. This is probably why it doesn't work for me because I'm not merging in this way.

Now my question is: does it work with repos with flat commits history? If not, then this should be documented.

**Comment by orta** [comment]
Yeah, this is why we've been recommending people use `danger pr` which is a lot more polished. You're welcome to take a look at fixing some of these issues, but TBH I'd expect `danger local` to end up having a different job soon enough https://github.com/danger/danger/issues/368

**Comment by orta** [comment]
Or to even make `--use-merged-pr` stop and say "you should use `danger pr`" or something

**Comment by php-coder** [comment]
> Yeah, this is why we've been recommending people use danger pr which is a lot more polished

Where do you recommend it? I follow getting started guide and run `danger init` that suggested to me to use `danger local`. Could you update this message then?

**Comment by orta** [comment]
Hrm, that might have been on the JS side, I should update this side - want to take a look at making it say `danger pr` instead of local in the [init command](https://github.com/danger/danger/blob/master/lib/danger/commands/init.rb#L266)?

**Comment by php-coder** [comment]
I would better to rely on you :)

**Comment by RishabhTayal** [comment]
Created a PR for this fix #796
